iree/hal/vulkan/vulkan_device.cc

```diff
@@ -249,8 +249,9 @@ StatusOr<ref_ptr<VulkanDevice>> VulkanDevice::Create(
   device_create_info.queueCreateInfoCount = queue_create_info.size();
   device_create_info.pQueueCreateInfos = queue_create_info.data();
   device_create_info.pEnabledFeatures = nullptr;
-  auto logical_device = make_ref<VkDeviceHandle>(
-      syms, enabled_device_extensions, /*allocator=*/nullptr);
+  auto logical_device =
+      make_ref<VkDeviceHandle>(syms, enabled_device_extensions,
+                               /*owns_device=*/true, /*allocator=*/nullptr);
```

Comment:
there's duplication here - it'd be nice to make this function (which creates the device and then picks the queues) do only what it needs to and defer to your new function for actually getting the queues from the device/etc.